torch/_inductor/codegen/triton.py

```diff
@@ -1904,6 +1904,21 @@ def codegen_node_schedule(self, node_schedule, numel, reduction_numel):
         kernel_name = self.define_kernel(src_code, node_schedule)
 
         kernel.call_kernel(V.graph.wrapper_code, kernel_name)
+
+        if config.generate_intermediate_hooks:
+            for node in node_schedule:
+                if not isinstance(node, scheduler.BaseSchedulerNode):
+                    continue
+                # TODO: not sure if this is the right thing to do
+                name = node.get_name()
+                if kernel.args.is_removed(name):
+                    continue
+                origin_node = node.node.get_origin_node()
+                if origin_node is not None:
+                    V.graph.wrapper_code.writeline(
+                        f"run_intermediate_hooks({origin_node.name!r}, {name})"
```

Comment:
fwiw - I tried taking the beefed up minifier for a spin on [issue](https://github.com/pytorch/pytorch/issues/100061), and hit an issue because it generated this code:
```
        buf262 = empty_strided((128, 10), (10, 1), device='cuda', dtype=torch.float32)
        buf263 = buf262; del buf262  # reuse
        buf264 = empty_strided((128, 10), (10, 1), device='cuda', dtype=torch.float32)
        triton_poi_fused_46.run(buf263, buf255, buf257, buf258, arg27_1, arg28_1, seed_cuda_0, buf264, 1280, grid=grid(1280), stream=stream0)
        run_intermediate_hooks('add_40', buf262)
```

It looks like `buf262` wasn't a true intermediate: I'm not familiar enough with the inductor code, but we immediately del it and create a `buf263` instead, which becomes the "real" intermediate.

Reply:
I'll take a look. You might be able to force it to work with `_inductor.config.inplace_buffers = False`

Reply:
https://github.com/pytorch/pytorch/pull/100322